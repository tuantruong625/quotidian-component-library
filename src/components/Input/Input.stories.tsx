import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { UserIcon } from '@heroicons/react/outline'
import Input from './Input'

export default {
 title: 'QuotidianComponentLibrary/⚛️ Atoms/Input',
 component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Text = Template.bind({})
Text.args = {
 type: 'text',
 placeholder: 'Placeholder',
 label: 'Text Input',
 id: 'hi'
}

export const Icon = () => {
 return (
  <Input type='text' label='Input with Icon' placeholder='Input with icon' icon={<UserIcon />} />
 )
}