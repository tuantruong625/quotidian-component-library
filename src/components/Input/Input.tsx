import React from "react"
import styled from "styled-components"
import { BorderRadius, Colors } from "../../utils"

export type InputProps = {
  type: string,
  placeholder: string,
  disabled?: boolean,
  shape?: "sm" | "md" | "lg" | "full",
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
}

type StyleProps = {
  shape?: "sm" | "md" | "lg" | "full",
}

const handleBorderRadius = (size?: string) => {
  switch (size) {
    case "sm":
      return `${BorderRadius.sm}`
    case "md":
      return `${BorderRadius.md}`
    case "lg":
      return `${BorderRadius.lg}`
    case "full":
      return `${BorderRadius.full}`
    default:
      return `${BorderRadius.none}`
  }
}

const QuotidianInput = styled.input<StyleProps>`
 padding: 0.5rem;
 border: 1px solid ${Colors.gray3};
 border-radius: ${({ shape }) => handleBorderRadius(shape)};
 color: ${Colors.gray4};

 &:disabled {
  cursor: not-allowed;
 }

 &::placeholder {
  color: ${Colors.gray3};
 }
`

const Input = ({ type = 'text', placeholder, disabled, shape, onChange, ...props }: InputProps): JSX.Element => {
  return (
    <QuotidianInput
      type={type}
      placeholder={placeholder}
      disabled={disabled}
      shape={shape}
      onChange={onChange}
      {...props}
    />
  )
}

export default Input