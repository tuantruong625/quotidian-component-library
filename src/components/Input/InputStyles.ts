import { size3, size2 } from './../../utils/spacing';
import { ReactElement } from "react"
import styled, { css } from "styled-components"
import { Colors, Spacing } from "../../utils"
import { handleBorderRadius } from "../utils"

type StyleProps = {
  shape?: "sm" | "md" | "lg" | "full",
  icon?: ReactElement<IconProps>,
}

export type IconProps = {
  height: string,
  width: string,
  color: string,
}

const debug = (item: any) => console.log(item);

export const QuotidianLabel = styled.label<StyleProps>`
  display: flex;
  justify-content: center;
  flex-direction: column;
  font-size: ${Spacing.size3};
  color: ${Colors.gray400};
  position: relative;
  
  > span:first-letter {
    text-transform: uppercase;
  }

  > svg {
      position: absolute;
      top: 1.5rem;
      margin: 0 ${Spacing.size2};
      color: ${Colors.success};
      width: 1em;
      height: 1em;
  }

  > input {
    padding: ${({ icon }) => icon ? `${Spacing.size2} ${Spacing.size2} ${Spacing.size2} ${Spacing.size5}` : `${Spacing.size2}`};
  }
`


export const QuotidianInput = styled.input<StyleProps>`
display: flex;
align-items: center;
padding: ${Spacing.size2};
border: 1px solid ${Colors.gray400};
border-radius: ${({ shape }) => handleBorderRadius(shape as string)};
color: ${Colors.gray400};

&:disabled {
  cursor: not-allowed;
}

&::placeholder {
  color: ${Colors.gray300};
}

&:focus {
  outline: none;
  border-color: ${Colors.primary1};
}
`

