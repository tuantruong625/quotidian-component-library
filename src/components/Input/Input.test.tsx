import React from "react";
import { render, screen } from "@testing-library/react";
import { UserIcon } from "@heroicons/react/outline";
import Input from "./Input";

const renderComponent = (props = {}) => {
 render(
  <Input
   type="text"
   placeholder="input with icon"
   label="Input with icon"
   {...props}
  />
 );
};
describe("Input", () => {
 it("renders an icon", () => {
  renderComponent({ icon: <UserIcon /> });

  expect(screen.getByLabelText("Input with icon")).toMatchSnapshot()
 });
});
