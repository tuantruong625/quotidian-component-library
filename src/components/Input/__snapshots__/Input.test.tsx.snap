// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders an icon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.5rem;
  border: 1px solid #4C566A;
  border-radius: 0px;
  color: #4C566A;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #D8DEE9;
}

.c0::-moz-placeholder {
  color: #D8DEE9;
}

.c0:-ms-input-placeholder {
  color: #D8DEE9;
}

.c0::placeholder {
  color: #D8DEE9;
}

.c0:focus {
  outline: none;
  border-color: #88C0D0;
}

<input
  class="c0"
  placeholder="input with icon"
  type="text"
/>
`;
